recipe/cmake-tests/CMakeLists.txt

```
@@ -0,0 +1,77 @@
+cmake_minimum_required(VERSION 3.17)
+
+project(diana)
+
+option(CUDA_FINDER "Which method find_package will use: CUDA or CUDAToolkit" "CUDA")
+option(WITH_ENABLE_LANGUAGE "Use enable_language(CUDA)" OFF)
+message(STATUS "WITH_ENABLE_LANGUAGE=${WITH_ENABLE_LANGUAGE} CUDA_FINDER=${CUDA_FINDER}")
+
+set(VALID_CUDA_FINDERS "CUDA" "CUDAToolkit" "OFF")
+if(NOT ${CUDA_FINDER} IN_LIST VALID_CUDA_FINDERS)
+    message(FATAL_ERROR "CUDA_FINDER must be one of: ${VALID_CUDA_FINDERS}")
+endif()
+
+file(GLOB cpu_source_files "${CMAKE_SOURCE_DIR}/src/*.cc")
+file(GLOB gpu_source_files "${CMAKE_SOURCE_DIR}/src/*.cu")
+file(GLOB include_files "${CMAKE_SOURCE_DIR}/src/*.hpp")
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED TRUE)
+
+if(WITH_ENABLE_LANGUAGE)
+    enable_language(CUDA)
+    set(CMAKE_CUDA_STANDARD 14)
+    set(CMAKE_CUDA_STANDARD_REQUIRED TRUE)
+    if(CUDA_FINDER)
+        find_package("${CUDA_FINDER}" REQUIRED)
+    endif(CUDA_FINDER)
+    add_definitions("-DWITHCUFILES")
+    add_executable(diana ${cpu_source_files} ${gpu_source_files})
+
+else(WITH_ENABLE_LANGUAGE)
+    add_executable(diana ${cpu_source_files})
+
+    if("${CUDA_FINDER}" STREQUAL "CUDA")
+        find_package("${CUDA_FINDER}" REQUIRED)
+        set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -Xcompiler ,\"-std=c++14\"")
+        add_definitions("-DWITHCUFILES")
+        CUDA_ADD_LIBRARY(diana_gpu ${gpu_source_files})
+        target_link_libraries(diana diana_gpu)
+    elseif("${CUDA_FINDER}" STREQUAL "CUDAToolkit")
+        # This example does not link to .cu libraries
+        find_package("${CUDA_FINDER}" REQUIRED)
+        target_link_libraries(diana CUDA::cudart)
+    endif("${CUDA_FINDER}" STREQUAL "CUDA")
+
+
+endif(WITH_ENABLE_LANGUAGE)
+
+
+# Some more checks
+if(WIN32)
+    set(NVCC_WRAPPER "$ENV{PREFIX}/Library/bin/nvcc.bat")
+else(WIN32)
+    set(NVCC_WRAPPER "$ENV{PREFIX}/bin/nvcc")
+endif(WIN32)
+
+if(CUDA_FINDER)
+    if("${CUDA_FINDER}" STREQUAL "CUDA")
+        set(FOUND_NVCC "${CMAKE_CUDA_COMPILER}")
```

I don't think this is valid. We'd want to test for `CMAKE_CUDA_COMPILER` in the case when `enable_language(CUDA)` is used, which is not every case of `find_package(CUDA)`. We should check the undocumented `CUDA_NVCC_EXECUTABLE` variable that is set here: https://gitlab.kitware.com/cmake/cmake/-/blob/v3.19.3/Modules/FindCUDA.cmake#L766-781

If we're not overriding it with our shim using an environment variable than we potentially should.
```suggestion
        set(FOUND_NVCC "${CUDA_NVCC_EXECUTABLE}")
```